palm/plugins/base_plugin_config.py

```diff
@@ -72,10 +72,10 @@ def validate(self, config: dict) -> bool:
 
     def _read(self) -> dict:
         palm_config = yaml.load(self.config_path.read_text(), Loader=yaml.FullLoader)
-        plugin_config = palm_config.get('plugin_config', {}).get(self.plugin_name, {})
+        plugin_config = palm_config.get("plugin_config", {}).get(self.plugin_name, {})
 
         if not plugin_config:
-            plugin_config = self.update()
+            self.update()
 
```

Comment:
This will fail because plugin_config is still None from above and read will (incorrectly) return None. 

I see why we need a patch here. But we need to change the return value of update, not the assignment back to plugin_config.

Can we also write a test around this?

Reply:
We also no longer need validate below. Because validate happens as part of update.

Last minute refactor bit me on this.

Reply:
So `update()` should also return the config dict?

Reply:
https://github.com/palmetto/palm-cli/pull/94